ddl/partition.go

```diff
@@ -129,7 +129,7 @@ func (w *worker) onAddTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (v
 		if tblInfo.TiFlashReplica != nil && tblInfo.TiFlashReplica.Available {
 			// For available state, the new added partition should wait it's replica to
 			// be finished. Otherwise the query to this partition will be blocked.
-			needWait, err := checkPartitionReplica(addingDefinitions, d)
+			needWait, err := checkPartitionReplica(tblInfo.TiFlashReplica.Count, addingDefinitions, d)
```

Comment:
If `needWait` is always true, it will also lead to a stuck situation. In this case, if we return an error, is TiFlash acceptable?

Reply:
I don't think it is correct. `needWait` will be true at the first time, then it becomes false after all tiflash regions ready. `needWait` does not mean `retry times`, it does not fit the commit title "let wait tiflash replica can exit when retry times beyond limitation".